assets/pages/regular-contributions/__tests__/ajaxTest.js

```diff
@@ -0,0 +1,96 @@
+import { getPaymentFields } from '../helpers/ajax';
+
+jest.mock('ophan', () => {});
+
+describe('Regular Contributions Payment fields', () => {
+
+  it('should create the correct payment field to handle direct debit', () => {
+    const sortCode = '200000';
+    const accountNumber = '55779911';
+    const accountHolderName = 'Bart Simpson';
+    const paymentFieldName = 'directDebitData';
+    const userId = '123456';
+
+    const expectedPaymentFields = {
+      accountHolderName,
+      sortCode,
+      accountNumber,
+    };
+    const paymentFields = getPaymentFields(
+      undefined,
+      accountNumber,
+      sortCode,
+      accountHolderName,
+      paymentFieldName,
+      userId,
+    );
+
+    expect(paymentFields.accountHolderName).toEqual(expectedPaymentFields.accountHolderName);
+    expect(paymentFields.sortCode).toEqual(expectedPaymentFields.sortCode);
```

Comment:
For all of these tests you could check that the function returns exactly the right number of fields, eg. 5 for DD, 1 for PP & 2 for Stripe

Reply:
Good call

Reply:
@rupertbates for DD should be 3 only.